docs/client-concepts/high-level/indexing/pipelines.asciidoc

```diff
@@ -0,0 +1,108 @@
+:ref_current: https://www.elastic.co/guide/en/elasticsearch/reference/6.x
+
+:github: https://github.com/elastic/elasticsearch-net
+
+:nuget: https://www.nuget.org/packages
+
+////
+IMPORTANT NOTE
+==============
+This file has been generated from https://github.com/elastic/elasticsearch-net/tree/6.x/src/Tests/Tests/ClientConcepts/HighLevel/Indexing/Pipelines.doc.cs. 
+If you wish to submit a PR for any spelling mistakes, typos or grammatical errors for this file,
+please modify the original csharp file found at the link and submit the PR with that change. Thanks!
+////
+
+[[ingest-pipelines]]
+=== Ingest Pipelines
+
+Elasticsearch will automatically re-route index requests to ingest nodes,
+however with some careful consideration you can optimise this path.
+
+[source,csharp]
+----
+public class Person
+{
+    public int Id { get; set; }
+    public string FirstName { get; set; }
+    public string LastName { get; set; }
+}
+----
+
+==== Create an ingestion pipeline
+
+Assuming we are indexing Person documents, we can create an ingestion pipeline that manipulates the
+incoming values before they are indexed.
+
+Lets assume that our application always expects surnames to be capitalised, and for initials to
+be indexed into their own field.
+
+We could acheive this requirement by creating a custom mapping and creating an ingest pipeline.
+The Person type can then be used as-is, without making any changes.
+
+[source,csharp]
+----
+client.CreateIndex("people");
+
+client.Map<Person>(p => p
```

Comment:
Perhaps an example that takes `Person` IP address and uses `ingest-geoip` to populate another field?

Reply:
Nice idea. Will add now.

Reply:
Done.